docs/docs/features/middleware.md

```diff
@@ -73,9 +73,35 @@ func NewHumaAPI() huma.API {
 }
 ```
 
+### Errors
+
+If your middleware encounters an error, you can stop the processing of the next middleware or operation handler by skipping the call to `next` and writing an error response.
+
+The [`huma.WriteErr(api, ctx, status, message, ...error)`](https://pkg.go.dev/github.com/danielgtaylor/huma/v2#WriteErr) function can be used to write nice structured error responses which respect client-driven content negotiation for marshaling:
+
+```go title="code.go"
+func MyMiddleware(ctx huma.Context, next func(ctx huma.Context)) {
+	// If there is a query parameter "error=true", then return an error
+	if ctx.Query("error") == "true" {
+		huma.WriteErr(api, ctx, http.StatusInternalServerError,
```

Comment:
@danielgtaylor sorry, one question here. Where the `api` is supposed to be defined? I'm trying to create a middleware in a separate file so I'm not using

```
api.UseMiddleware(func(ctx huma.Context, next func(ctx huma.Context)) {
	[code with access to api]
})
```

Also how can I send parameters to the middleware? for example I have a `keysRetriever` app that I need to use inside the middleware. Thanks!